pentaho-proc-governor/impl/src/main/java/org/hitachivantara/process/ProcessGovernorImpl.java

```diff
@@ -83,22 +85,27 @@ private Runnable startProcess( String[] command, CompletableFuture<Process> futu
       Optional<Integer> exitValue = Optional.empty();
       try {
         semaphore.acquire();
-        logger.debug( "Executing command " + Arrays.toString( command ) );
+        if ( logger.isDebugEnabled() ) {
+          logger.debug( String.format( "Executing command %s", Arrays.toString( command ) ) );
+        }
         Process proc = getProcess( command );
         futureProc.complete( proc );
 
         exitValue = Optional.of( getExitValue( proc ) );
 
       } catch ( IOException | InterruptedException e ) {
+        Thread.currentThread().interrupt();
```

Comment:
wouldn't this prevent the next two lines from running?

Reply:
No, that will reset the interrupt status.  It's generally bad practice to catch interruptedexception and not reset the status, since things up the stack won't have a chance to respond to a request to interrupt.  Good overview:  https://www.yegor256.com/2015/10/20/interrupted-exception.html

Java interrupt handling is seriously wack.